packages/hoppscotch-common/src/components/mockServer/CreateMockServer.vue

```diff
@@ -0,0 +1,513 @@
+  mockServers$,
+  addMockServer,
+  updateMockServer as updateMockServerInStore,
+} from "~/newstore/mockServers"
+import { restCollections$ } from "~/newstore/collections"
+import { TippyComponent } from "vue-tippy"
+import {
+  createMockServer as createMockServerMutation,
+  updateMockServer,
+} from "~/helpers/backend/mutations/MockServer"
+import { WorkspaceType } from "~/helpers/backend/graphql"
+import { copyToClipboard as copyToClipboardHelper } from "~/helpers/utils/clipboard"
+import { refAutoReset } from "@vueuse/core"
+import { pipe } from "fp-ts/function"
+import * as TE from "fp-ts/TaskEither"
+
+// Icons
+import IconServer from "~icons/lucide/server"
+import IconPlay from "~icons/lucide/play"
+import IconSquare from "~icons/lucide/square"
+import IconCopy from "~icons/lucide/copy"
+import IconCheck from "~icons/lucide/check"
+import MockServerLogs from "~/components/mockServer/MockServerLogs.vue"
+
+const t = useI18n()
+const toast = useToast()
+
+// Modal state
+const modalData = useReadonlyStream(showCreateMockServerModal$, {
+  show: false,
+  collectionID: undefined,
+  collectionName: undefined,
+})
+
+const mockServers = useReadonlyStream(mockServers$, [])
+const collections = useReadonlyStream(restCollections$, [])
+
+// Component state
+const mockServerName = ref("")
+const loading = ref(false)
+const showCloseButton = ref(false)
+const createdServer = ref<any>(null)
+const delayInMsVal = ref<string>("0")
+const isPublic = ref<boolean>(true)
+const showLogs = ref(false)
+const selectedCollectionID = ref("")
+const selectedCollectionName = ref("")
+const tippyActions = ref<TippyComponent | null>(null)
+
+// Props computed from modal data
+const show = computed(() => modalData.value.show)
+const collectionID = computed(() => modalData.value.collectionID)
+const collectionName = computed(
+  () => modalData.value.collectionName || "Unknown Collection"
+)
+
+// Find existing mock server for this collection
+const existingMockServer = computed(() => {
+  if (!collectionID.value) return null
+  return mockServers.value.find(
+    (server) => server.collectionID === collectionID.value
+  )
+})
+
+const isExistingMockServer = computed(() => !!existingMockServer.value)
+
+// Collection options for the selector
+const collectionOptions = computed(() => {
+  const flattenCollections = (collections: any[], prefix = ""): any[] => {
+    const result: any[] = []
+
+    collections.forEach((collection) => {
+      const displayName = prefix
+        ? `${prefix} / ${collection.name}`
+        : collection.name
+      result.push({
+        label: displayName,
+        value: collection.id || collection._ref_id,
+        collection: collection,
+      })
+
+      // Add folders as nested options
+      if (collection.folders && collection.folders.length > 0) {
+        result.push(...flattenCollections(collection.folders, displayName))
+      }
+    })
+
+    return result
+  }
+
+  return flattenCollections(collections.value)
+})
```

Comment:
Are we foreseeing performance bottlenecks with the collection selector for large collection trees?